File: 2610-convert-an-array-into-a-2d-array-with-conditions/2610-convert-an-array-into-a-2d-array-with-conditions.cpp
class Solution {
public:
    vector<vector<int>> findMatrix(vector<int>& nums) {
        unordered_map<int, int> count;
        vector<vector<int>> res;

        for (int n : nums) {
            int row = count[n]; //no. of arrays needed
            if (res.size() == row) {
                res.push_back({});
            }

            res[row].push_back(n); //add curr integer to array
            count[n]++;
        }

        return res;
    }
};

File: 2610-convert-an-array-into-a-2d-array-with-conditions/2610-convert-an-array-into-a-2d-array-with-conditions.py
class Solution(object):
    def findMatrix(self, nums):
        """
        :type nums: List[int]
        :rtype: List[List[int]]
        """
        count = defaultdict(int)
        res = []
        
        for n in nums:
            row = count[n]
            if len(res) == row:
                res.append([])
                
            res[row].append(n)
            count[n] += 1
        return res
        